articles/openshift/oshift-hawkular-metrics.md

```diff
@@ -0,0 +1,75 @@
+---
+title: How to use the Hawkular Metrics API | UKCloud Ltd
+description: Shows how to structure basic queries to the Hawkular Metrics API in order to get resource statistics back
+services: openshift
+author: Kieran O'Neill
+
+toc_rootlink: How To
+toc_sub1: 
+toc_sub2:
+toc_sub3:
+toc_sub4:
+toc_title: Hawkular Metrics
+toc_fullpath: How To/oshift-how-create-users.md
+toc_mdlink: oshift-hawkular-metrics.md
+---
+
+# How to use the Hawkular Metrics API
+
+## Overview
+
+This guide is intended to give you enough information to forumulate basic memory usage or cpu queries against nodes in your cluster. For more advanced usage please refer to the Hawkular documentation at: https://www.hawkular.org/hawkular-metrics/docs/user-guide/#_metrics or the OpenShift Hawkular repository readme at https://github.com/openshift/origin-metrics/blob/master/docs/hawkular_metrics.adoc
+
+## Using the API
+
+1. Firstly, you must ensure that the user you will be using to query the API has reader permissions for the corresponding OpenShift project to the Hawkular-Tenant you are querying. If you are querying the _system tenant then the user must have cluster-reader permissions.
+
+2. To get a list of all node level counters and gauges you can use you would run the following:
+    ```
+    curl -X GET https://hawkular-metrics.$(cluster-url)/hawkular/metrics/metrics "Content-Type: application/json" -H "Hawkular-Tenant: _system" -H "Authorization: Bearer $token"
+    ```
+
+    Where $token is a variable storing the output of oc whoami -t from an OpenShift user with the correct permissions to view metrics and cluster-url is the domain suffix of your cluster.
+    
+    For example; if your cluster URL is https://ocp.openshift-test.cna.ukcloud.com:8443 your query would be the following:
+    ```
+    curl -X GET https://hawkular-metrics.openshift-test.cna.ukcloud.com/hawkular/metrics/metrics "Content-Type: application/json" -H "Hawkular-Tenant: _system" -H "Authorization: Bearer $token"
+    ```
+
+3. You will need to pass the output from the request into some form of json parser in order to get human readable results. It will return a series of json blocks similiar to the following:
+```
```

Comment:
Indent each line of the code sample by four spaces so that it indents correctly when rendered